app.js

```diff
@@ -34,26 +31,42 @@ function lockCards(cardsToLock) {
 		card.removeEventListener('click', handleUserInput);
 	});
 	firstCard = null;
+	pausePlay = false;
+
+
 	console.log('someone called lockCards,Cards match');
+
 };
 
 function handleUserInput() {
-	this.classList.add('turned'); // you always want to do this.
+	if (pausePlay) return;
+	this.classList.add('turned');
+	if (this === firstCard) return
 	if (!firstCard) {
-		firstCard = this; //if the firstCard didn't exist, set the first card
+		firstCard = this;
 	} else {
-		currentCard = this; //now it's time to compare both cards!
+		currentCard = this;
+
 		isEqual(firstCard, currentCard);
+
 		isTheSameCard ? lockCards([firstCard, currentCard]) : resetCards([firstCard, currentCard]);
+
 	}
 }
 
 function resetCards(cardsToReset) {
+	pausePlay = true;
 	cardsToReset.forEach((card) => {
 		setTimeout(() => {
-			card.classList.remove('turned')
+			pausePlay = false;
```

Comment:
for the sake of maintainability, you should encapsulate this behaviour in functions.

i.e.

enablePlay();
resumePlay();

All these functions would do is set `pausePlay`  to `true` or `false`, but if you ever want to modify the implementation, you don't have to touch the code in 10 different  places, only 1.

Reply:
Understood